src/JsonParseNode.php

```diff
@@ -282,7 +282,21 @@ public function getDateTimeValue(): ?DateTime {
      * @throws Exception
      */
     public function getDateIntervalValue(): ?DateInterval{
-        return ($this->jsonNode !== null) ? new DateInterval(strval($this->jsonNode)) : null;
+        if ($this->jsonNode === null){
+            return null;
+        }
+        $originalValue = strval($this->jsonNode);
+        $negativeValPosition = strpos($originalValue, '-');
+        $invert = 0;
+        $str = $originalValue;
+        if ($negativeValPosition !== false && $negativeValPosition === 0) {
+            // Invert the interval
+            $invert = 1;
+            $str = substr($originalValue, 1);
+        }
+        $dateInterval = new DateInterval($str);
```

Comment:
What happens if $str is not a valid date interval -, we could do a check here:
`php
if (@DateInterval::createFromDateString($dateIntervalSpec) === false) {
        throw new InvalidArgumentException('jsonNode must be a valid date interval specification');
    }
`
That way, we are certain this line uses a valid value for str and prevents unforseen bugs

Reply:
I have intentionally left out the invalid case since that will, by default, throw an exception.